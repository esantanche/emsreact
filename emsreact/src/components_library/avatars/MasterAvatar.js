import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import Avatar from "@material-ui/core/Avatar";

/**
 * Avatar for lists of articles.
 *
 * @param {string} label This is a label for ARIA (Accessible Rich Internet Applications)
 * @param {string} color Color for the avatar
 * @param {string} letter Letter for the avatar
 */
class MasterAvatar extends Component {

    static propTypes = {
        label: PropTypes.string,
        color: PropTypes.string,
        letter: PropTypes.string
    };

    render() {

        const { label, color, letter } = this.props;

        return (

            <Avatar aria-label={label} style={{ backgroundColor: color }}>
                {letter}
            </Avatar>

        )
    }

}

export default MasterAvatar;

