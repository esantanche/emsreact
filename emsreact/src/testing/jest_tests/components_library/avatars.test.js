// FIXME docs

// import ReactDOM from "react-dom";
import React from "react";
import renderer from 'react-test-renderer';
import MasterAvatar from "../../../components_library/avatars/MasterAvatar";

import red from "@material-ui/core/colors/red";

// Here we test components in components_library/avatars

it('MasterAvatar renders correctly', () => {
    const tree = renderer
        .create(<MasterAvatar color={red[500]}
                              letter="W"
                              label="How I work" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});