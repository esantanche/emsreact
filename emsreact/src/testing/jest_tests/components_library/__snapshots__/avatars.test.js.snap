// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasterAvatar renders correctly 1`] = `
<div
  aria-label="How I work"
  className="MuiAvatar-root-1 MuiAvatar-colorDefault-2"
  style={
    Object {
      "backgroundColor": "#f44336",
    }
  }
>
  W
</div>
`;
